jwst/srctype/srctype.py

```diff
@@ -45,6 +47,10 @@ def set_source_type(input_model):
             src_type = 'EXTENDED'
             log.info('Exposure is a background target; setting SRCTYPE = %s' % src_type)
 
+        elif pipe_utils.is_tso(input_model):
```

Comment:
I'm surprised that the user-supplied value (if any) doesn't have higher priority.  That's OK; I'm just surprised.